Various updates.

davidlehn opened the pull request.

davidlehn committed:
Various updates.

- Switch to a module.
- Update dependencies.
- Simplify dependencies.
- Remove currently unused testing support.
- Update linting rules.
- Ignore default output files.
- Cleanup repo template files.
- Update to use newer cborld.

davidlehn committed:
I/O updates.

- Support "-" for stdin input and stdout output.
- Only print output message in verbose mode.

davidlehn committed:
Print out compact JSON byte comparison.

davidlehn committed:
Update actions.

- Test on Node.js 20.x.
- Update action verions.
- Disable tests.

dlongley commented on package.json:
```diff
@@ -17,60 +30,25 @@
-    "@digitalbazaar/cborld": "^4.0.0",
-    "esm": "^3.2.22",
-    "yargs": "^15.3.1"
+    "@digitalbazaar/cborld": "^5.2.0",
```
6.0.1 now out.

davidlehn committed:
Add more docs.

davidlehn committed:
Update dependencies.

BigBlueHat commented on README.md:
````diff
@@ -78,7 +79,17 @@ npx @digitalbazaar/cborld-cli --help
 ```
-TBD
+# basic conversion of file, with verbose and diagnostic output
+./cborld encode --verbose --diagnose ../cborld/examples/note.jsonld -o note.cborld
+
+# decode file
+./cborld decode --verbose --diagnose ./note.cborld
+
+# output to hex string
+./cborld encode input.jsonld -o - | xxd -p -c 0 -
+
+# decode hex string
+xxd -r -p -c 0 input.hex | ./cborld decode -v -d -
````
Local name is `cbor-cli.js`, but globally installed name is `cborld`, so the above doesn't work.

But this does!
```suggestion
cborld encode --verbose --diagnose ../cborld/examples/note.jsonld -o note.cborld

# decode file
cborld decode --verbose --diagnose ./note.cborld

# output to hex string
cborld encode input.jsonld -o - | xxd -p -c 0 -

# decode hex string
xxd -r -p -c 0 input.hex | cborld decode -v -d -
```

BigBlueHat replied:
Actually...use this one. It makes each command runnable in succession. We may also want to bring in the `note.jsonld` file into this repo to avoid downloading more stuff.

```suggestion
cborld encode --verbose --diagnose ../cborld/examples/note.jsonld -o note.cborld

# decode file
cborld decode --verbose --diagnose ./note.cborld
## ...will generate `out.jsonld`

# output to hex string
cborld encode out.jsonld -o - | xxd -p -c 0 - > out.hex

# decode hex string
xxd -r -p -c 0 out.hex | cborld decode -v -d -
## ...will generate `out.jsonld`
```

BigBlueHat commented on lib/index.js:
```diff
@@ -1,8 +1,275 @@
+  return appMaps;
+}
+
+// Yars command to decode given CBOR-LD bytes to a JSON-LD document
```
```suggestion
// Yargs command to decode given CBOR-LD bytes to a JSON-LD document
```

BigBlueHat reviewed (changes requested):
One more wee nit.

davidlehn committed:
Fix typo.

Co-authored-by: BigBlueHat <byoung@bigbluehat.com>

davidlehn committed:
Update dev dependencies.

davidlehn committed:
Update to `@digitalbazaar/cborld@7`.

- **BREAKING**: This has various codec changes.

davidlehn committed:
Update changelog.

davidlehn committed:
Update actions.

davidlehn committed:
Update description.

davidlehn committed:
Fix distributed files.

davidlehn committed:
Update examples.

Co-authored-by: BigBlueHat <byoung@bigbluehat.com>

davidlehn committed:
Update README.

- Fix bugs.
- Improve instructions.
- Use local examples.
- Update .gitignores and .gitattributes.
